pkg/jobrunaggregator/jobtableprimer/job_name_generator.go

```diff
@@ -0,0 +1,248 @@
+				break
+			}
+		}
+		for _, releaseConfigURL := range s.releaseConfigURLs {
+			if strings.Contains(releaseConfigURL, release.Release) {
+				found = true
+				break
+			}
+		}
+		if !found {
+			newReleases = append(newReleases, release.Release)
+		}
+	}
+	for _, release := range newReleases {
+		s.addReleaseURLs(release)
+	}
+}
+
+func (s *jobNameGenerator) GenerateJobNames() ([]string, error) {
+	jobNames := []string{}
+
+	for _, url := range s.releaseConfigURLs {
+		resp, err := http.Get(url)
+		if err != nil {
+			return jobNames, fmt.Errorf("error reading %v: %w", url, err)
+		}
+		if resp.StatusCode < 200 || resp.StatusCode > 299 {
+			return jobNames, fmt.Errorf("error reading %v: %v", url, resp.StatusCode)
+		}
+
+		content, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return jobNames, fmt.Errorf("error reading %v: %w", url, err)
+		}
+		resp.Body.Close()
+
+		releaseConfig := &FakeReleaseConfig{}
+		if err := json.Unmarshal(content, releaseConfig); err != nil {
+			return jobNames, fmt.Errorf("error reading %v: %w", url, err)
+		}
+
+		jobNames = append(jobNames, fmt.Sprintf("// begin %v", url))
+		localLines := []string{}
+		for _, curr := range releaseConfig.Verify {
+			localLines = append(localLines, curr.ProwJob.Name)
+		}
+		sort.Strings(localLines)
+		jobNames = append(jobNames, localLines...)
+		jobNames = append(jobNames, fmt.Sprintf("// end %v", url))
+		jobNames = append(jobNames, "")
+	}
+
+	for _, url := range s.periodicURLs {
+		resp, err := http.Get(url)
+		if err != nil {
+			return jobNames, fmt.Errorf("error reading %v: %w", url, err)
+		}
+		if resp.StatusCode < 200 || resp.StatusCode > 299 {
+			return jobNames, fmt.Errorf("error reading %v: %v", url, resp.StatusCode)
+		}
+
+		content, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return jobNames, fmt.Errorf("error reading %v: %w", url, err)
+		}
+		resp.Body.Close()
+
+		periodicConfig := &FakePeriodicConfig{}
+		if err := yaml.Unmarshal(content, periodicConfig); err != nil {
+			return jobNames, fmt.Errorf("error reading %v: %w", url, err)
+		}
+
+		jobNames = append(jobNames, fmt.Sprintf("// begin %v", url))
+		localLines := []string{}
+		for _, curr := range periodicConfig.Periodics {
+			// TODO: the single file for say "master" actually contains every release, but we only want jobs 4.10+
+			// where we started disruption monitoring. Adding a bunch of future rows to buy us time but this could
+			// stand some logic.
+			if !strings.Contains(curr.Name, "-4.10") &&
+				!strings.Contains(curr.Name, "-4.11") &&
+				!strings.Contains(curr.Name, "-4.12") &&
+				!strings.Contains(curr.Name, "-4.13") &&
+				!strings.Contains(curr.Name, "-4.14") &&
+				!strings.Contains(curr.Name, "-4.15") &&
+				!strings.Contains(curr.Name, "-4.16") &&
+				!strings.Contains(curr.Name, "-4.17") &&
+				!strings.Contains(curr.Name, "-4.18") &&
+				!strings.Contains(curr.Name, "-4.19") &&
+				!strings.Contains(curr.Name, "-4.20") {
+				continue
+			}
```

Comment:
Does this get changed so we don't have to update anymore?  Can we check for all releases > 4.10 and check contains against ReleaseRow.Release?

Reply:
Good catch! I guess I just copied the code without seeing the release string dealing there. :)